v2/pkg/genruntime/secrets/collector.go

```diff
@@ -8,33 +8,38 @@ package secrets
 import (
 	"sort"
 
+	"github.com/pkg/errors"
 	"golang.org/x/exp/maps"
 	v1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kerrors "k8s.io/apimachinery/pkg/util/errors"
 
 	"github.com/Azure/azure-service-operator/v2/pkg/genruntime"
 )
 
-// SecretCollector collects secret values and their associated genruntime.SecretDestination's
+// TODO: This is similar to configmaps.Collector. If this is updated, that should likely be as well
```

Comment:
maybe we should consider extract `Collector` interface and extend it here and for configmaps?

Reply:
Not sure if something like type `Collector[T v1.Secret|v1.ConfigMap] struct {}` would work? Kinda new to generics, but would be clean if we can use generics here.

Reply:
I thought about using generics here, but the whole `StringData`/`BinaryData` and how secret writes to `StringData` show up in data as base64 encoded, etc means the two have different application functions, so the type ends up needing to be something like:

```
type Collector[T metav1.Object] struct {
	things  map[string]*T
	namespace string
	applierFunc func(t T) error
}
```

or something like that.

We could do it, and it would save a small bit of code mostly the `Values` implementation, but the actual meat of `AddValue` couldn't be shared, so I decided against it at least for now. If we get a 3rd impl then maybe it makes sense to generalize, but with 2 I'm not sure that it does.

Reply:
We'd also need to define an interface other than `metav1.Object` because there's no shared way to get/set values from `SecretData`/`BinaryData` somehow (not in the v1 package though so can't add methods to v1.ConfigMap/v1.Secret to do this easily...).